src/mixins/date-mixin.ts

```diff
@@ -58,17 +56,20 @@ export default class DateMixin extends Mixins(CommonMixin) {
   }
 
   /**
-   * Creates and returns a new Date object in UTC, given parameters in Pacific timezone in string format YYYY-mm-dd.
+   * Converts a date string (YYYY-MM-DD) to a Date object at 12:00:00 am Pacific time.
+   * @example 2021-11-22 -> 2021-11-22T08:00:00.00Z
    */
-  createUtcDateFromStr (dateStr: string): Date {
-    // safety check
+  yyyyMmDdToDate (dateStr: string): Date {
```

Comment:
thanks for cleaning this function up!

Reply:
It turns out you and I both implemented this (in different projects). I took the best of both code :)